gobblin-core/src/main/java/gobblin/converter/filter/AvroSchemaFieldRemover.java

```diff
@@ -99,6 +99,9 @@ private Schema removeFields(Schema schema, Map<String, Schema> schemaMap) {
   }
 
   private Schema removeFieldsFromRecords(Schema schema, Map<String, Schema> schemaMap) {
+    Schema newRecord = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.isError());
```

Comment:
I chatted with @zliu41 and the current approach should be fine since the schema is no longer a member field and it is used only in the context of removeFields() recursion.

LGTM with @zliu41  adding some code comments about the situation.